### DISCUSSION TITLE
Error handling policy

### DISCUSSION THREAD

**Discussion by multimeric** [umccr/BioSerDe#9] [Polls]
We will be interfacing with a lot of disparate libraries in this project. The two best practise crates I know of for error handling are [`anyhow`](https://github.com/dtolnay/anyhow) and [`thiserror`](https://github.com/dtolnay/thiserror). `anyhow` is definitely easier to use, but results in a less well-defined API, since all your error types are effectively a boxed error that you can't do anything with except panic or ignore. `thiserror` gives a nicely defined API, but at the expense of a lot of maintenance burden.

In theory `thiserror` is a good fit here, but to lessen the coding burden I'm happy to work with `anyhow` instead.

Let me know which you would prefer.

**Comment by brainstorm** [comment]
I've experience with both, but for this type of project I think that `thiserror` is a better fit, happy to be convinced of the opposite though.

**Reply by multimeric** [reply]
I agree it's a better fit, but it's a big burden to use.

**Reply by brainstorm** [reply]
Happy to support issues that might arise with `thiserror`, we used it with close to 0 problems in htsget-rs, i.e: https://github.com/umccr/htsget-rs/blob/main/htsget-http-core/src/error.rs
